src/epmdless_client.erl

```diff
@@ -90,17 +90,17 @@
 -record(node_key, {
           %% using e-mail terminology: https://www.w3.org/Protocols/rfc822/#z8
           %% addr-spec   =  local-part "@" domain        ; global address
-          local_part :: atom(),
-          domain :: atom()
+          local_part :: atom() | string(),
+          domain :: atom() | string()
          }).
 
 -record(node, {
-          key :: #node_key{},
-          name_atom :: atom(), %% formatted full node name as an atom
-          addr :: inet:ip_address(),
-          host :: inet:hostname(),
-          port :: inet:port_number(),
-          added_ts :: integer()
+          key       :: undefined | #node_key{},
+          name_atom :: '_' |'$3' | atom(), %% formatted full node name as an atom
```

Comment:
Did you add these based on what `dialyzer` has spitted out? Where are these single atoms coming from?

Reply:
from macro `NODE_MATCH`